src/elements-experimental/seat-reservation/seat-reservation-area/seat-reservation-area.component.ts

```diff
@@ -0,0 +1,66 @@
+import { forceType } from '@sbb-esta/lyne-elements/core/decorators.js';
+import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
+import { html, LitElement } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+import { type AreaBackgroundState, type ElementMounting } from '../seat-reservation.js';
+
+import style from './seat-reservation-area.scss?lit&inline';
+
+/**
+ * Visualize an area with a special meaning within a wagon.
+ */
+export
+@customElement('sbb-seat-reservation-area')
+class SbbSeatReservationAreaElement extends LitElement {
+  public static override styles: CSSResultGroup = style;
+
+  /** Rotation Prop */
+  @forceType()
+  @property({ attribute: 'rotation', type: Number })
+  public accessor rotation: number = 0;
+
+  /** Width Prop */
+  @forceType()
+  @property({ attribute: 'width', type: Number })
+  public accessor width: number = 6;
+
+  /** Height Prop */
+  @forceType()
+  @property({ attribute: 'height', type: Number })
+  public accessor height: number = 6;
+
+  /** Mounting Prop */
+  @forceType()
+  @property({ attribute: 'mounting', type: String })
+  public accessor mounting: ElementMounting = 'FREE';
+
+  /** Background Prop */
+  @forceType()
+  @property({ attribute: 'background', type: String })
+  public accessor background: AreaBackgroundState = 'LIGHT';
+
+  protected override willUpdate(changedProperties: PropertyValues<this>): void {
+    super.willUpdate(changedProperties);
+    if (changedProperties.has('width')) {
+      this.style?.setProperty('--sbb-reservation-area-width', `${this.width}`);
+    }
+    if (changedProperties.has('height')) {
+      this.style?.setProperty('--sbb-reservation-area-height', `${this.height}`);
+    }
+    if (changedProperties.has('rotation')) {
+      this.style?.setProperty('--sbb-reservation-area-rotation', `${this.rotation}`);
+    }
+  }
+
+  protected override render(): TemplateResult {
+    return html`<span class="sbb-sr-area"><slot></slot></span>`;
```

Comment:
Potentially it makes sense to remove the span and add all styles to the host. then the CSS variables for width, height and rotation would become obsolete, as an outer element can just overwrite the dimensions.
```suggestion
    return html`<slot></slot>`;
```

Reply:
see https://github.com/sbb-design-systems/lyne-components/pull/3750#discussion_r2120939326